### PR TITLE
Refine platform::Transform function and fix prelu_op testing.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/Paddle/issues/4202
Fix https://github.com/PaddlePaddle/Paddle/issues/4204

Thanks for @hedaoyuan's help.

### CODE REVIEW

**Review by hedaoyuan — APPROVED** [review]

File: paddle/platform/transform.h [hunk 1]
```diff
@@ -29,45 +29,71 @@
...
+                  BinaryOperation op) {
+    auto place = context.GetPlace();
+    PADDLE_ENFORCE(is_cpu_place(place), "It must use CPU place.");
     std::transform(first1, last1, first2, result, op);
-  } else {
+  }
+};
+
 #ifdef __NVCC__
+template <>
```
Comment (jacquesqiao) [inline paddle/platform/transform.h]: 为什么这样改一下就不会有问题了呢

**Review by reyoung — CHANGES_REQUESTED** [review]
I think this #4209 is much simpler.

File: paddle/platform/transform.h (hunk 1 above)
Comment (qingqing01) [reply paddle/platform/transform.h]: @jacquesqiao 改了之后，带了模板参数，对于CPUPlace 、 GPUPlace编译出来是两个不同的符号。

原先那样，在.cu文件里**链接**时，用的是第一次g++/gcc编译同出来的同名符号，就出现问题。

**Comment by hedaoyuan** [comment]
Such as the [comment in the pr 4209](https://github.com/PaddlePaddle/Paddle/pull/4209/files#r139862597), this code is more efficient after compile.

**Review by reyoung — APPROVED** [review]

**Pull request merged**